File: src/file.rs
use std::io::Write;
use std::{
    fs::{File, OpenOptions},
    io::{BufRead, BufReader},
};

const RESULTS_FILE: &str = "results.txt";

pub(crate) fn read_results() -> (i32, i32) {
    if let Ok(file) = File::open(RESULTS_FILE) {
        let reader = BufReader::new(file);
        for line in reader.lines().map_while(Result::ok) {
            let parts: Vec<&str> = line.trim().split(',').collect();
            if parts.len() == 2 {
                if let (Ok(last_number), Ok(max_divisors)) = (parts[0].parse(), parts[1].parse()) {
                    return (last_number, max_divisors);
                }
            }
        }
    }
    (0, 0) // Default start values if file is missing or malformed
}

pub(crate) fn write_results(number: i32, max_divisors: i32) {
    let mut file: File = OpenOptions::new()
        .write(true)
        .create(true)
        .truncate(true)
        .open(RESULTS_FILE)
        .unwrap();
    writeln!(file, "{},{}", number, max_divisors).unwrap();
}

File: src/File.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.StandardOpenOption;

public class File {
    private static final String RESULTS_FILE = "results.txt";

    public static ResultPair readResults() {
        Path path = Path.of(RESULTS_FILE);

        if (Files.exists(path)) {
            try (BufferedReader reader = Files.newBufferedReader(path)) {
                String line;
                while ((line = reader.readLine()) != null) {
                    String[] parts = line.trim().split(",");
                    if (parts.length == 2) {
                        try {
                            int lastNumber = Integer.parseInt(parts[0]);
                            int maxDivisors = Integer.parseInt(parts[1]);
                            return new ResultPair(lastNumber, maxDivisors);
                        } catch (NumberFormatException ignored) {
                            // TODO: Skip malformed line
                        }
                    }
                }
            } catch (IOException e) {
                System.out.println(e.getMessage());
            }
        }

        // Default if file is missing or malformed
        return new ResultPair(0, 0);
    }

    public static void writeResults(int number, int maxDivisors) {
        Path path = Path.of(RESULTS_FILE);

        try (BufferedWriter writer = Files.newBufferedWriter(path,
                StandardOpenOption.CREATE,
                StandardOpenOption.TRUNCATE_EXISTING,
                StandardOpenOption.WRITE)) {

            writer.write(number + "," + maxDivisors);
            writer.newLine();

        } catch (IOException e) {
            System.out.println(e.getMessage());
        }
    }
}
